components/qr-reader.js

```diff
@@ -0,0 +1,61 @@
+import { Html5QrcodeScanner, Html5QrcodeScanType } from 'html5-qrcode';
+import { Fingerprint, Webcam } from 'lucide-react';
+import { toast } from 'react-toastify';
+
+export default function QrReader({ setDID }) {
+    let scanner;
+
+    function startCamera() {
+        scanner = new Html5QrcodeScanner(
+            'reader',
+            {
+                fps: 8,
+                qrbox: {
+                    width: 250,
+                    height: 250,
+                },
+                experimentalFeatures: {
+                    useBarCodeDetectorIfSupported: true,
+                },
+                rememberLastUsedCamera: true,
+                supportedScanTypes: [Html5QrcodeScanType.SCAN_TYPE_CAMERA],
+            },
+            false
+        );
+        if (scanner) {
+            scanner.clear();
+            scanner.render(onScanSuccess, onScanFailure);
+        }
+    }
+
+    async function onScanSuccess(decodedResult) {
+        scanner.clear();
+        const result = extractQRData(decodedResult);
+
+        console.log(`scanned result: ${result}`);
+        setDID(result);
+        toast.success('QR Code Scanned Successfully');
+        return result;
+    }
+
+    const extractQRData = (decodedResult) => decodedResult;
```

Comment:
I'm confused by this function. Should a TODO be added there or it can be deleted?

Reply:
```suggestion

```

Reply:
Was this suggestion tested? I see `extractQRData` is used on line 33. This suggestion is just removing that function so accepting it would result in an error. I suggest taking all Suggested changes and doing a commit with them but before that make sure the code is tested

Reply:
Yeah you are right. That part make no sense since we are extracting directly the Qr code result. I just removed and leave it like 

```javascript
async function onScanSuccess(decodedResult) {
        scanner.clear();
        const result = decodedResult;

        console.log(`scanned result: ${result}`);
        setDID(result);
        toast.success('QR Code Scanned Successfully');
        return result;
    }
```